cypress/integration/SidePanelLabs.spec.ts

```diff
@@ -0,0 +1,82 @@
+import * as h from '../helpers';
+
+describe('Side Panel', () => {
+  before(() => {
+    h.stories.visit();
+  });
+
+  ['Default', 'AsAside', 'AsDiv'].forEach(story => {
+    context(`given the "${story}" story is rendered`, () => {
+      beforeEach(() => {
+        h.stories.load('Testing|React/Labs/Side Panel/Cypress', story);
+      });
+
+      const name = /Accessible Label Name/i;
+
+      it('should not have any axe errors', () => {
+        cy.checkA11y();
+      });
+
+      context('the button', () => {
+        it(`should have an accessible name`, () => {
+          cy.findByRole('button', {name}).should('exist');
+        });
+
+        it(`should have an aria-expanded attribute of 'true'`, () => {
+          cy.findByRole('button', {name}).should('have.attr', 'aria-expanded', 'true');
+        });
+
+        it(`should have an aria-controls attribute equal to the id of the panel`, () => {
+          cy.findByRole('button', {name}).then(button => {
+            const buttonAriaControlsValue = button.attr('aria-controls');
+            switch (story) {
+              case 'AsAside':
+                cy.findByRole('complementary', {name}).should(
+                  'have.attr',
+                  'id',
+                  buttonAriaControlsValue
+                );
+                break;
+              default:
+                cy.findByRole('region', {name}).should('have.attr', 'id', buttonAriaControlsValue);
+            }
+          });
+        });
+      });
+
+      it(`should have a panel with a landmark role`, () => {
+        switch (story) {
+          case 'AsAside':
+            cy.findByRole('complementary', {name}).should('exist');
+            break;
+          default:
+            cy.findByRole('region', {name}).should('exist');
+        }
+      });
+
+      context('when tabbing', () => {
+        beforeEach(() => {
+          cy.tab();
+        });
+
+        context('the button', () => {
+          it('should be focused', () => {
+            cy.findByRole('button', {name}).should('have.focus');
+          });
+        });
+      });
```

Comment:
This one is a little weird. There is no explicit starting point for focus.

What's the point of it? Are we specifying that the button should be focusable? The modal specs use `cy.tab` after some action where there is a specification for where focus should go and that focus is controlled within the modal, but nothing on the original trigger button.

If the point is to specify that the button should be focusable, maybe we should add a `.should('be.focusable')` that runs checks to verify that the element in question can receive focus explicitly. As this is written, focusability is indirectly measured through `cy.tab`. But without a starting point, `cy.tab` is over verifying.

Reply:
We talked about this one offline and created a new story for Cypress that showcases a hypothetical "header" with an Avatar button that could be the last focusable element before going to the Side Panel. When a user hits tab _once_ that should then put focus onto the control button of the panel despite it having other elements within it.